dll/win32/shell32/folders/CRecycleBin.cpp

```diff
@@ -171,16 +123,94 @@ BOOL WINAPI CBSearchRecycleBin(IN PVOID Context, IN HANDLE hDeletedFile)
     return ret;
 }
 
-static PIDLRecycleStruct * _ILGetRecycleStruct(LPCITEMIDLIST pidl)
+/*
+ * Recycle Bin folder
+ */
+
+HRESULT CRecyclerExtractIcon_CreateInstance(
+    LPCITEMIDLIST pidl, REFIID riid, LPVOID * ppvOut)
 {
-    LPPIDLDATA pdata = _ILGetDataPointer(pidl);
+    PIDLRecycleStruct *pFileDetails = _ILGetRecycleStruct(pidl);
```

Comment:
I think you can just bail out quickly if `pFileDetails` happens to be NULL.
(And I know that some of the code here does the check, and some other doesn't and this is buggy, especially when it then dereferences the pointer.)

Reply:
I think that function expects a root, and in case it's null it uses the default recycled, i can specify this, obtaining it from the file, tomorrow i will do that